src/app/components/media/CreateMediaSource.js

```diff
@@ -0,0 +1,493 @@
+import React from 'react';
+import Relay from 'react-relay/classic';
+import { FormattedMessage, FormattedHTMLMessage } from 'react-intl';
+import Box from '@material-ui/core/Box';
+import IconButton from '@material-ui/core/IconButton';
+import KeyboardArrowDown from '@material-ui/icons/KeyboardArrowDown';
+import Card from '@material-ui/core/Card';
+import CardHeader from '@material-ui/core/CardHeader';
+import CardContent from '@material-ui/core/CardContent';
+import Collapse from '@material-ui/core/Collapse';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import Button from '@material-ui/core/Button';
+import Typography from '@material-ui/core/Typography';
+import TextField from '@material-ui/core/TextField';
+import CancelIcon from '@material-ui/icons/Cancel';
+import AddCircleOutlineIcon from '@material-ui/icons/AddCircleOutline';
+import LinkifyIt from 'linkify-it';
+import { makeStyles } from '@material-ui/core/styles';
+import Message from '../Message';
+import UploadFile from '../UploadFile';
+import CreateSourceMutation from '../../relay/mutations/CreateSourceMutation';
+import SourcePicture from '../source/SourcePicture';
+import globalStrings from '../../globalStrings';
+import { getErrorObjects, getErrorMessage } from '../../helpers';
+import CheckError from '../../CheckError';
+import GenericUnknownErrorMessage from '../GenericUnknownErrorMessage';
+import {
+  Row,
+  units,
+  StyledIconButton,
+} from '../../styles/js/shared';
+import {
+  StyledAvatarEditButton,
+  StyledTwoColumns,
+  StyledSmallColumn,
+  StyledBigColumn,
+  StyledName,
+} from '../../styles/js/HeaderCard';
+
+
+const useStyles = makeStyles(theme => ({
+  headerRow: {
+    display: 'flex',
+    alignItems: 'top',
+    justifyContent: 'space-between',
+    marginBottom: theme.spacing(2),
+  },
+  CancelSaveRow: {
+    textAlign: 'right',
+  },
+}));
+
+
+function CreateMediaSource({
+  media,
+  onCancel,
+  relateToExistingSource,
+}) {
+  const [expandName, setExpandName] = React.useState(true);
+  const [expandAccounts, setExpandAccounts] = React.useState(true);
+  const [sourceName, setSourceName] = React.useState('');
+  const [primaryUrl, setPrimaryUrl] = React.useState({ url: '', error: '' });
+  const [submitDisabled, setSubmitDisabled] = React.useState(true);
+  const [dialogOpen, setDialogOpen] = React.useState(false);
+  const [validate_primary_link_exist, setValidatePrimaryLinkExist] = React.useState(true);
```

Comment:
Please change to `validatePrimaryLinkExist`

Reply:
Done